### PR TITLE
[mds-agency] Improve error handling when POSTing a vehicle

### PR DESCRIPTION
**Pull request by avatarneil**
## PR Checklist

 - [x] simple searchable title - `[mds-db] Add PG env var`, `[config] Fix eslint config`
 - [x] briefly describe the changes in this PR
 - [ ] mark as draft if should not be merged
 - [ ] write tests for all new functionality

## Impacts
- [ ] Provider
- [x] Agency
- [ ] Audit
- [ ] Policy
- [ ] Compliance

### CODE REVIEW

File: packages/mds-agency/api.ts [hunk 1]
```diff
@@ -316,73 +316,51 @@ function api(app: express.Express): express.Express {
...
+      await writeRegisterEvent()
+      res.status(201).send({ result: 'register device success', recorded, device })
+    } catch (err) {
+      if (String(err).includes('duplicate')) {
+        res.status(409).send({
+          error: 'already_registered',
+          error_description: 'A vehicle with this device_id is already registered'
+        })
+      } else if (String(err).includes('db')) {
+        res.status(500).send(new ServerError())
```
Comment (karcass) [inline packages/mds-agency/api.ts]: why no log in this block but log in next block?

File: packages/mds-agency/api.ts [hunk 2]
```diff
@@ -316,73 +316,51 @@ function api(app: express.Express): express.Express {
...
-          })
-        } else {
-          log.error(providerName(res.locals.provider_id), 'register vehicle failed:', err).then(() => {
-            res.status(500).send(new ServerError())
-          })
-        }
+    try {
+      await db.writeDevice(device)
+      await Promise.all([cache.writeDevice(device), stream.writeDevice(device)])
+      device.status = VEHICLE_STATUSES.removed
```
Comment (mdurling) [inline packages/mds-agency/api.ts]: The device has already been written to db/cache/stream .. what does setting the status here accomplish? In general, I would prefer `res.status(201).send({ result: 'register device success', recorded, device: { ...device, status: VEHICLE_STATUSES.removed })`

File: packages/mds-agency/api.ts [hunk 3]
```diff
@@ -316,73 +316,51 @@ function api(app: express.Express): express.Express {
...
+        telemetry: null,
+        timestamp: recorded,
+        trip_id: null,
+        recorded,
+        telemetry_timestamp: undefined,
+        service_area_id: null
+      }
+      try {
+        await db.writeEvent(event)
+        await Promise.all([cache.writeEvent(event), stream.writeEvent(event)]).catch(async err => {
```
Comment (mdurling) [inline packages/mds-agency/api.ts]: This could be within a `try/catch` instead of using `Promise.catch`, right?

Comment (karcass) [reply packages/mds-agency/api.ts]: done

File: packages/mds-agency/api.ts (hunk 2 above)
Comment (karcass) [reply packages/mds-agency/api.ts]: probably that mutation should happen at the top

**Pull request merged**